butterfly/ImageLayer/HDF5.py

```diff
@@ -19,3 +19,14 @@ def load_tile(query):
             vol = fd[fd.keys()[0]]
             return vol[z0:z1:Sk,y0:y1:Sj,x0:x1:Si]
 
+    @staticmethod
+    def preload_source(query):
```

Comment:
Here's `preload_source` in the hdf5 file. It opens the path for the `TileQuery` instance. It returns the full size of a single slice as the blocksize and the type of the data as a string.